### PR TITLE
Update RTD installation instructions

### PR DESCRIPTION
**Pull request by hbushouse**
This PR addresses an issue raised in Help Desk ticket [INC0196726](https://stsci.service-now.com/nav_to.do?uri=incident.do%3Fsys_id=d88e80c49733b15023a2fe971153af66%26sysparm_stack=incident_list.do%3Fsysparm_query=active=true) where there was no mention of the fact that we are currently limited to installing jwst with python<=3.11. So I've updated all the places where "conda create" is mentioned.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: docs/getting_started/install.rst [hunk 1]
```diff
@@ -58,9 +64,9 @@ Installing Previous Releases
 
 You can also install a specific version (from `jwst 0.17.0` onward):
 
-    | >> conda create -n <env_name> python
+    | >> conda create -n <env_name> python=3.11
     | >> conda activate <env_name>
-    | >> pip install jwst==1.3.3
+    | >> pip install jwst==1.12.5
```
Comment (nden) [inline docs/getting_started/install.rst:69]: Do you also want to update the section below this one, about specific versions before jwst 0.17.0. Or should we remove this section?

**Review by nden — APPROVED** [review]

File: docs/getting_started/install.rst (hunk 1 above)
Comment (hbushouse) [reply docs/getting_started/install.rst:69]: I'm tempted to remove it, because it's so old and obsolete now.

Comment (hbushouse) [reply docs/getting_started/install.rst:69]: removed in latest commit

**Comment by jdavies-st** [comment]
I note that you are successfully testing under 3.12 now in the 1.13.x tagged versions and on master.  Will Python 3.12 eventually be officially supported once one of those 1.13.x releases makes it to pypi?

**Pull request merged**